testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2183,6 +2198,82 @@ public void testTrinoAnalyze()
         onTrino().executeQuery("DROP TABLE " + trinoTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
+    public void testRegisterTableSuccess(StorageFormat storageFormat)
+            throws TException
+    {
+        String baseTableName = "test_register_table_success_" + storageFormat.name().toLowerCase(ENGLISH) + "_" + randomTableSuffix();
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onTrino().executeQuery(format("CREATE TABLE %s (a int, b varchar, c boolean) with (format = '%s', partitioning = ARRAY['a'])", trinoTableName, storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s values(1, 'INDIA', true)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s values(2, 'USA', false)", trinoTableName));
+        onTrino().executeQuery(format("COMMENT ON TABLE %s IS '%s'", trinoTableName, "This is table comment"));
+        onTrino().executeQuery(format("COMMENT ON COLUMN %s.%s IS '%s'", trinoTableName, "b", "This is column comment"));
+
+        List<Row> expected = List.of(row(1, "INDIA", true), row(2, "USA", false));
+
+        assertThat(onTrino().executeQuery(format("SELECT * FROM %s", trinoTableName)))
+                .containsOnly(expected);
+        assertThat(onSpark().executeQuery(format("SELECT * FROM %s", sparkTableName)))
+                .containsOnly(expected);
+
+        String describeSparkOldTable = (String) onSpark().executeQuery("DESCRIBE TABLE EXTENDED " + sparkTableName).toString();
+        String showCreateTrinoOldTable = (String) onTrino().executeQuery("SHOW CREATE TABLE " + trinoTableName).toString();
+        String tableLocation = getTableLocation(trinoTableName, true);
+
+        // Drop table from hive metastore and use the same table name to register again with the metadata
+        metastoreClient.dropTable(TEST_SCHEMA_NAME, baseTableName, false);
+        assertFalse(metastoreClient.getAllTables(TEST_SCHEMA_NAME).contains(baseTableName));
+
+        onTrino().executeQuery(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableName, tableLocation));
+        assertThat(onTrino().executeQuery(format("SELECT * FROM %s", trinoTableName)))
+                .containsOnly(expected);
+        assertThat(onSpark().executeQuery(format("SELECT * FROM %s", sparkTableName)))
+                .containsOnly(expected);
+
+        String describeSparkNewTable = (String) onSpark().executeQuery("DESCRIBE TABLE EXTENDED " + sparkTableName).toString();
```

Comment:
The output of `onSpark().executeQuery(...)` is a `QueryResult`.

Apply `row(0).get(0)` and cast to `String` to get the content you want.

Reply:
`row(0).get(0)` -> `Iterables.getOnlyElement( getOnlyElement ( ) )`

Reply:
`onSpark().executeQuery(...).getOnlyValue()`

This helper method is available now as well.

Reply:
Addressed

Reply:
`DESCRIBE TABLE EXTENDED` returns multiple lines/rows. Do we have any helper method other than toString() to convert it into a string?

Reply:
Used the below statement to make multiline comparison

                 assertThat(actualShowCreateTable)
                .hasColumns(expectedShowCreateTable.getColumnTypes())
                .containsExactlyInOrder(expectedShowCreateTableRows);